### PR TITLE
agent: remove BDS

### PR DESCRIPTION
**Pull request by kyessenov**
Change-Id: Ie3e87863e4113b9b667dff3eb2282296e8e3c097

Removes BDS _experimental_ feature since it failed to be promoted.
CC @zirain

### CODE REVIEW

**Comment by keithmattix** [comment]
I honestly didn't know we had this feature. Do we not have anyone using it? Was it documented?

**Comment by kyessenov** [comment]
The flag was needed for `applyTo: BOOTSTRAP` in https://istio.io/latest/docs/reference/config/networking/envoy-filter/, and was mostly needed for Wasm bootstrap extensions. The replacement is using a custom bootstrap template instead of the built-in, which can solve all sorts of bootstrap issues and has been around since the beginning.

**Comment by zirain** [comment]
I recall there's a e2e test for this, but it looks like it was removed earlier.

File: releasenotes/notes/bds-removal.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: bug-fix
```
Comment (hanxiaop) [inline releasenotes/notes/bds-removal.yaml:2]:
```suggestion
kind: feature
```

**Review by hzxuzhonghu — COMMENTED** [review]
LGTM since it is a very alpha feature

**Review by zirain — APPROVED** [review]
LGTM

users can use custom bootstrap instead

**Comment by howardjohn** [comment]
Does this break EnvoyFilter BOOTSTRAP mode? If so, can we make sure we mark
it deprecated?

Do we also need to cleanup the Istiod implementation?

**Comment by kyessenov** [comment]
@howardjohn Good point. Removed istiod code and sending a PR to api with a deprecation note.

**Review by howardjohn — APPROVED** [review]

**Comment by sschepens** [comment]
@kyessenov we used this feature to add custom `stat_tags` regexes to our proxies, is there another way to do this? MeshConfig only allows for customizing included metrics but not custom stat tags.

**Comment by kyessenov** [comment]
@sschepens You can mount a bootstrap template and make any customizations in a template. Alternatively, we can extend mesh config / proxy config - is this for Wasm?

**Comment by sschepens** [comment]
@kyessenov I created an issue #50894 that's the real problem, we were mitigating this by patching bootstrap config

**Pull request merged**